From 3142ef6efb8b88a2408c9164a88144a13774b764 Mon Sep 17 00:00:00 2001
From: Aleksei Nikiforov <aleksei.nikiforov@linux.ibm.com>
Date: Wed, 26 Jun 2024 12:15:44 +0200
Subject: [PATCH] Link libc10 to pthreads

It gets linked with libmkl on amd64,
but it's explicitely required on s390x

Linking issue only appears when using
gcc-13 with gold linker.
---
 c10/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/c10/CMakeLists.txt b/c10/CMakeLists.txt
index 3d78e94aed6bd..80e172497d5e6 100644
--- a/c10/CMakeLists.txt
+++ b/c10/CMakeLists.txt
@@ -125,6 +125,10 @@ if(NOT BUILD_LIBTORCHLESS)
     add_dependencies(c10 mimalloc-static)
   endif()
 
+  if(LINUX)
+    target_link_libraries(c10 PRIVATE Threads::Threads)
+  endif()
+
   if(ANDROID)
     target_link_libraries(c10 PRIVATE log)
   endif()
